xbmc/addons/AddonManager.cpp

```diff
@@ -1195,15 +1222,31 @@ bool CAddonMgr::AddonsFromRepoXML(const CRepository::DirInfo& repo, const std::s
     if (info)
     {
       CAddonBuilder builder;
-      auto basePath = URIUtils::AddFileToFolder(repo.datadir, info->identifier);
+      std::string path, directory;
+      auto metadata = CServiceBroker::GetAddonMgr().GetExtension(info, "xbmc.addon.metadata");
+      if (!metadata)
+        metadata = CServiceBroker::GetAddonMgr().GetExtension(info, "kodi.addon.metadata");
+      if (metadata)
+        path = CServiceBroker::GetAddonMgr().GetExtValue(metadata->configuration, "path");
+
+      if (path.empty())
+        directory = info->identifier;
+      else
+        directory = StringUtils::Split(path, '/')[0];
+
+      auto basePath = URIUtils::AddFileToFolder(repo.datadir, directory);
```

Comment:
Not that big of a deal but can't you refactor `Factory` so this whole can be moved there? So many edge cases spread all over now. I don't think the 'info->plugin_path' hack right below is needed either anymore.

Reply:
@tamland please have a look at the modified implementation of path attribute

Reply:
looks good. thanks!